src/measure_theory/function/convergence_in_measure.lean

```diff
@@ -0,0 +1,272 @@
+begin
+  intros ε hε,
+  specialize h_tendsto ε hε,
+  suffices : (λ i, μ {x | ε ≤ dist (f' i x) (g' x)}) = (λ i, μ {x | ε ≤ dist (f i x) (g x)}),
+    by rwa this,
+  refine funext (λ i, measure_congr _),
+  filter_upwards [h_left i, h_right],
+  intros x hxf hxg,
+  rw eq_iff_iff,
+  change ε ≤ dist (f' i x) (g' x) ↔ ε ≤ dist (f i x) (g x),
+  rw [hxg, hxf],
+end
+
+lemma congr_left (h : ∀ i, f i =ᵐ[μ] f' i) (h_tendsto : tendsto_in_measure μ f g) :
+  tendsto_in_measure μ f' g :=
+h_tendsto.congr h (eventually_eq.rfl)
+
+lemma congr_right (h : g =ᵐ[μ] g') (h_tendsto : tendsto_in_measure μ f g) :
+  tendsto_in_measure μ f g' :=
+h_tendsto.congr (λ i, eventually_eq.rfl) h
+
+end tendsto_in_measure
+
+section
+
+variables [metric_space E]
+variables {f : ℕ → α → E} {g : α → E}
+
+/-- Auxiliary lemma for `tendsto_in_measure_of_tendsto_ae`. -/
+lemma tendsto_in_measure_of_tendsto_ae_of_measurable
+  [measurable_space E] [second_countable_topology E] [borel_space E] [is_finite_measure μ]
+  (hf : ∀ n, measurable (f n)) (hg : measurable g)
+  (hfg : ∀ᵐ x ∂μ, tendsto (λ n, f n x) at_top (𝓝 (g x))) :
+  tendsto_in_measure μ f g :=
+begin
+  refine λ ε hε, ennreal.tendsto_at_top_zero.mpr (λ δ hδ, _),
+  by_cases hδi : δ = ∞,
+  { simp only [hδi, implies_true_iff, le_top, exists_const], },
+  lift δ to ℝ≥0 using hδi,
+  rw [gt_iff_lt, ennreal.coe_pos, ← nnreal.coe_pos] at hδ,
+  obtain ⟨t, htm, ht, hunif⟩ := tendsto_uniformly_on_of_ae_tendsto' hf hg hfg hδ,
+  rw ennreal.of_real_coe_nnreal at ht,
+  rw metric.tendsto_uniformly_on_iff at hunif,
+  obtain ⟨N, hN⟩ := eventually_at_top.1 (hunif ε hε),
+  refine ⟨N, λ n hn, _⟩,
+  suffices : {x : α | ε ≤ dist (f n x) (g x)} ⊆ t, from (measure_mono this).trans ht,
+  rw ← set.compl_subset_compl,
+  intros x hx,
+  rw [set.mem_compl_eq, set.nmem_set_of_eq, dist_comm, not_le],
+  exact hN n hn x hx,
+end
+
+/-- Convergence a.e. implies convergence in measure in a finite measure space. -/
+lemma tendsto_in_measure_of_tendsto_ae
+  [measurable_space E] [second_countable_topology E] [borel_space E] [is_finite_measure μ]
+  (hf : ∀ n, ae_measurable (f n) μ) (hg : ae_measurable g μ)
+  (hfg : ∀ᵐ x ∂μ, tendsto (λ n, f n x) at_top (𝓝 (g x))) :
+  tendsto_in_measure μ f g :=
+begin
+  refine tendsto_in_measure.congr (λ i, (hf i).ae_eq_mk.symm) hg.ae_eq_mk.symm _,
+  refine tendsto_in_measure_of_tendsto_ae_of_measurable (λ i, (hf i).measurable_mk)
+    hg.measurable_mk _,
+  have hf_eq_ae : ∀ᵐ x ∂μ, ∀ n, (hf n).mk (f n) x = f n x,
+    from ae_all_iff.mpr (λ n, (hf n).ae_eq_mk.symm),
+  filter_upwards [hf_eq_ae, hg.ae_eq_mk, hfg],
+  intros x hxf hxg hxfg,
+  rw [← hxg, funext (λ n, hxf n)],
+  exact hxfg,
+end
+
+/-- If `f` is a sequence of functions which converges in measure to `g`, then there exists a
+subsequence of `f` which converges a.e. to `g`. -/
+lemma tendsto_in_measure.exists_seq_tendsto_ae
+  (hfg : tendsto_in_measure μ f g) :
+  ∃ ns : ℕ → ℕ, ∀ᵐ x ∂μ, tendsto (λ i, f (ns i) x) at_top (𝓝 (g x)) :=
```

Comment:
can you also record in the conclusion that `ns` is `strict_mono`? (Otherwise, `ns` could be a constant, who knows?!)